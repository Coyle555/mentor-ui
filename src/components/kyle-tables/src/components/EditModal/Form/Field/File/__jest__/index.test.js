jest.mock('mentor-inputs', () => {
	return { FileInput: props => <div>{JSON.stringify(props)}</div> };
});

import React from 'react';
import { FileField } from '../index';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';

afterEach(cleanup);

test('File field render with value', () => {
	const tree = renderer.create(<FileField value="/foo" />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('File field render with no value', () => {
	const tree = renderer.create(<FileField />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('File field deleting a file', () => {
	const onDeleteClick = jest.fn();

	const { getByText } = render(
		<FileField
			fieldId="bar"
			onDeleteClick={onDeleteClick}
			rowId="baz"
			value="/foo"
		/>
	);

	fireEvent.click(getByText('Delete File'));
	expect(onDeleteClick).toHaveBeenCalledWith('baz', 'bar');
});

