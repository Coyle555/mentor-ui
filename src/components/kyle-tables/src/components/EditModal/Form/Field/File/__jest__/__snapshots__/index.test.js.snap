// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File field render with no value 1`] = `
<div
  className="file-input-container"
>
  <div
    className="file-input"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div>
      {"label":"Upload File"}
    </div>
  </div>
</div>
`;

exports[`File field render with value 1`] = `
<div
  className="file-input-container"
>
  <div
    className="file-display"
  >
    <a
      className="file-link"
      download={true}
      href="/foo"
    >
      File
    </a>
    <p
      className="file-delete"
      onClick={[Function]}
    >
      Delete File
    </p>
  </div>
  <div
    className="file-input"
    style={
      Object {
        "width": "75%",
      }
    }
  >
    <div>
      {"label":"Upload File"}
    </div>
  </div>
</div>
`;
